### DISCUSSION TITLE
Possible to add a System after the world is configured?

### DISCUSSION THREAD

**Discussion by RefuX** [Quillraven/Fleks#144] [General]
Thinking about integrating Fleks into my project. Playing around with the API and looks like I can only add systems at world configuration time is this correct?

Thanks!

**Comment by Quillraven** [comment]
That is correct but you can disable a system to ignore it during a world update. You can enable and disable a system at any time.

What is your use-case to add a system later on? Would not be a huge issue to add that but I am interested to understand why the current approach would not work for you.

**Reply by RefuX** [reply]
Sure, so my game has modding.
I could scan all the mods looking for systems and then put placeholders in during world creation. Or, as it seems (to me) more straightforward to add the mod's systems when the mod is loaded, and remove the mod's systems when the mod is unloaded.

**Reply by Quillraven** [reply]
OK, but do you need a special order of those systems? Or do you just want to add them at the end/beginning?

Also, keep in mind that when you create a system later on and it is linked to a family it can result in a performance spike because the family also needs to be created and initialized with its entities.

I don't know your game but if you have like 100 of thousands of entities then this will most likely end up in a short "lag" when creating and adding the system.

**Reply by RefuX** [reply]
I was just thinking about the ordering issue! I think the simplest solution would be for the API to let the user define at which index to add the system. Since I can see which systems are currently installed, I can then reason about where to place it.

**Reply by Quillraven** [reply]
I will have a look at it but don't expect it today or tomorrow, sorry 😅

If you want to help: the world currently has an array of systems for better iteration performance. Also, a system has a default value for the world which is the world that currently gets configured.

So a possible solution could be to add new functions to the world to add/remove systems with and without index. This will require an array copy I guess.

Would be cleaner and easier to have an ArraList for systems but needs to be verified with the benchmark files if that has any noticeable impact. Or maybe there is even a better data structure for that.

If you are interested and motivated feel free to make a PR 😊

Otherwise, I will have a look next weekend most likely.

**Reply by RefuX** [reply]
I've cloned the repo and will have a poke about  😎
